src/gui/widgets/SimpleTextFloat.cpp

```diff
@@ -45,13 +45,44 @@ SimpleTextFloat::SimpleTextFloat() :
 
 	m_textLabel = new QLabel(this);
 	layout->addWidget(m_textLabel);
+
+	m_showTimer = new QTimer();
+	m_showTimer->setSingleShot(true);
+	QObject::connect(m_showTimer, &QTimer::timeout, this, &SimpleTextFloat::show, Qt::AutoConnection);
+
+	m_hideTimer = new QTimer();
+	m_hideTimer->setSingleShot(true);
+	QObject::connect(m_hideTimer, &QTimer::timeout, this, &SimpleTextFloat::hide, Qt::AutoConnection);
 }
 
 void SimpleTextFloat::setText(const QString & text)
 {
 	m_textLabel->setText(text);
 }
 
+void SimpleTextFloat::showWithDelay(int msecBeforeDisplay, int msecDisplayTime)
+{
+	if (msecBeforeDisplay != 0)
+	{
+		m_showTimer->start(msecBeforeDisplay);
+	}
+	else
+	{
+		show();
+	}
```

Comment:
Could just write ``msecBeforeDisplay != 0 ? m_showTimer->start(msecBeforeDisplay) : show();``, although this is more personal preference. Keep the current code if you prefer, I just like the ternary version because it's one line.

Reply:
I prefer well block defined.